### PR TITLE
hide forbidden errs from kubectl get all output

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

Related downstream bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1515703

Since we do not know if the alias `all` maps to a resource that
a user cannot list, ignore forbidden errors when that alias is specified

cc @fabianofranz @soltysh

### CODE REVIEW

**Comment by soltysh** [comment]
@juanvallejo the code itself lgtm, would you mind adding test case for it, please?

**Comment by juanvallejo** [comment]
@soltysh thanks, added a small test, ptal

**Comment by juanvallejo** [comment]
@soltysh friendly ping

File: pkg/kubectl/cmd/resource/get.go [hunk 1]
```diff
@@ -181,6 +182,10 @@ func (options *GetOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args
 		return nil
 	}
 
+	if len(args) > 0 && args[0] == "all" {
```
Comment (fabianofranz) [inline pkg/kubectl/cmd/resource/get.go]: There's no guarantee that's where "all"  will be coming in the list of args, users could do e.g. `kubectl get nodes,all`. Also, we want to remove the references to hardcoded "all" on the client, it's now a server-side concept that the client doesn't have detailed knowledge about.

**Comment by fabianofranz** [comment]
This should probably happen at the builder level. You know you have to ignore forbidden errors on resources you got from a category expander. Note that you can still have valid/wanted forbidden errors for a single `get` call, if the forbidden resource came somehow other than the category expander. E.g. `kubectl get all,nodes` should still do what it does today if you don't have access to `nodes`.

File: pkg/kubectl/cmd/resource/get.go (hunk 1 above)
Comment (soltysh) [reply pkg/kubectl/cmd/resource/get.go]: Hmm... that's fair, but how else are you proposing to go with this?

Comment (soltysh) [reply pkg/kubectl/cmd/resource/get.go]: Nvmd, just noticed the next comment ;)

**Comment by juanvallejo** [comment]
Thanks will update to have the resource builder implicitly ignore forbidden errors from a category expander

**Comment by juanvallejo** [comment]
@fabianofranz thanks for the feedback, updated patch to ignore forbidden errors from expanded resources at the builder level, will update test if this approach works, PTAL

**Comment by juanvallejo** [comment]
@fabianofranz updated to test cases such as `kubectl get all,nodes,...`

**Comment by juanvallejo** [comment]
There was a small discussion started on the bugzilla regarding potential usability concerns that a fix like this one might introduce https://bugzilla.redhat.com/show_bug.cgi?id=1515703#c7.

While it could be argued that hiding forbidden errors returned for resources expanded from an alias (such as `all`) could hide confusing errors from a user, it also hides information that could be useful in determining why certain resources are not showing up.

For example, given a set of resources that are part of an alias `foo`, and a subset of those resources that return forbidden errors when listed, it is easier to determine that 
  a) those resources are supposed to be listed as part of the `foo` alias, and 
  b) they could not be listed because the current user did not have permission to fetch those resources specifically.

Given the scenario that this patch would introduce, a user A would receive a list of resources as part of alias `foo`, but a user B would only receive a partial list of resources for that same alias. User B would have an expectation of seeing list of resource types that are supposed to be listed as part of alias `foo`, but would not know immediately why all resources are not being listed (if they are not authorized to list a specific subset).

Given the last scenario, I believe that hiding forbidden errors from a user (whether they come from a resource kind expanded from an alias), has the  potential to introduce more confusion. Instead, I believe that the actual problem to address in the linked bugzilla [1] is to allow users to list cronjob resources in the online cluster.

1. https://bugzilla.redhat.com/show_bug.cgi?id=1515703#c7

**Pull request closed without merging**